libraries/Zigbee/src/ZigbeeHandlers.cpp

```diff
@@ -0,0 +1,148 @@
+/* Zigbee Common Functions */
+#include "ZigbeeCore.h"
+#include "Arduino.h"
+
+#if SOC_IEEE802154_SUPPORTED
+
+// forward declaration of all implemented handlers
+static esp_err_t zb_attribute_set_handler(const esp_zb_zcl_set_attr_value_message_t *message);
+static esp_err_t zb_attribute_reporting_handler(const esp_zb_zcl_report_attr_message_t *message);
+static esp_err_t zb_cmd_read_attr_resp_handler(const esp_zb_zcl_cmd_read_attr_resp_message_t *message);
+static esp_err_t zb_configure_report_resp_handler(const esp_zb_zcl_cmd_config_report_resp_message_t *message);
+static esp_err_t zb_cmd_default_resp_handler(const esp_zb_zcl_cmd_default_resp_message_t *message);
+
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+
+// Zigbee action handlers
+static esp_err_t zb_action_handler(esp_zb_core_action_callback_id_t callback_id, const void *message) {
+  esp_err_t ret = ESP_OK;
+  switch (callback_id) {
+    case ESP_ZB_CORE_SET_ATTR_VALUE_CB_ID:          ret = zb_attribute_set_handler((esp_zb_zcl_set_attr_value_message_t *)message); break;
+    case ESP_ZB_CORE_REPORT_ATTR_CB_ID:             ret = zb_attribute_reporting_handler((esp_zb_zcl_report_attr_message_t *)message); break;
+    case ESP_ZB_CORE_CMD_READ_ATTR_RESP_CB_ID:      ret = zb_cmd_read_attr_resp_handler((esp_zb_zcl_cmd_read_attr_resp_message_t *)message); break;
+    case ESP_ZB_CORE_CMD_REPORT_CONFIG_RESP_CB_ID:  ret = zb_configure_report_resp_handler((esp_zb_zcl_cmd_config_report_resp_message_t *)message); break;
+    case ESP_ZB_CORE_CMD_DEFAULT_RESP_CB_ID:        ret = zb_cmd_default_resp_handler((esp_zb_zcl_cmd_default_resp_message_t *)message); break;
+    default:                                 log_w("Receive unhandled Zigbee action(0x%x) callback", callback_id); break;
+  }
+  return ret;
+}
+
+#pragma GCC diagnostic pop
```

Comment:
Same here. Maybe use `[[maybe_unused]]`

Reply:
`warning: 'esp_err_t zb_action_handler(esp_zb_core_action_callback_id_t, const void*)' defined but not used [-Wunused-function]` Can you suggest the change?

Reply:
changed to your suggestion :)